### PR TITLE
SRV-35 inventory synchronization implementation

### PR DESCRIPTION
**Pull request by c3zi**

### CODE REVIEW

File: app/Repository/Supply/NetworkCampaignRepository.php [hunk 1]
```diff
@@ -20,19 +20,74 @@
...
+        $networkCampaign->time_end = $campaign->getDateEnd();
+
+        $networkCampaign->targeting_requires = $campaign->getTargetingRequires();
+        $networkCampaign->targeting_excludes = $campaign->getTargetingExcludes();
+
+        $networkCampaign->save();
+
+        foreach ($networkBanners as $banner) {
+            $networkCampaign->banners()->save($banner);
+        }
```
Comment (PawelPodkalicki) [inline app/Repository/Supply/NetworkCampaignRepository.php]: Lines 
```
foreach ($networkBanners as $banner) {
            $networkCampaign->banners()->save($banner);
}
```
could be swapped with
`$networkCampaign->banners()->saveMany($networkBanners);`

If accepted, remember about test `testImportWhenActivateIsSuccessful`
```
        $repository
            ->expects($this->exactly(2))
            ->method('save');
```

File: app/Models/NetworkCampaign.php [hunk 2]
```diff
@@ -44,13 +44,19 @@ class NetworkCampaign extends Model
...
+        'targeting_excludes' => 'json',
+    ];
+
     /**
      * The attributes that are mass assignable.
      *
      * @var array
      */
     protected $fillable = [
         'uuid',
```
Comment (teghnet) [inline app/Models/NetworkCampaign.php:58]: I would call everything `id` & `parent_id`

File: src/Supply/Application/Service/InventoryImporter.php [hunk 3]
```diff
@@ -18,52 +18,63 @@
...
 {
-    /*** @var DemandClient */
-    private $client;
+    /** @var MarkedCampaignsAsDeleted */
+    private $markedCampaignsAsDeletedService;
 
     /** @var CampaignRepository */
     private $campaignRepository;
 
+    /*** @var DemandClient */
```
Comment (PawelPodkalicki) [inline src/Supply/Application/Service/InventoryImporter.php]:
```suggestion
    /** @var DemandClient */
```

File: scripts/install.sh [hunk 4]
```diff
@@ -8,6 +8,7 @@ mkdir -pm 777 ${INSTALLATION_PATH}/storage
 
 # Move directories
 mv app ${INSTALLATION_PATH}/
+mv src ${INSTALLATION_PATH}/
```
Comment (teghnet) [inline scripts/install.sh]: duplicate

File: src/Common/Domain/Adapter/ArrayCollection.php [hunk 5]
```diff
@@ -18,7 +18,7 @@
  * along with AdServer. If not, see <https://www.gnu.org/licenses/>
  */
 
-declare(strict_types = 1);
+declare(strict_types=1);
```
Comment (teghnet) [inline src/Common/Domain/Adapter/ArrayCollection.php]: formatting

File: src/Common/Domain/ValueObject/Uuid.php [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with AdServer. If not, see <https://www.gnu.org/licenses/>
+ */
+
+namespace Adshares\Common\Domain\ValueObject;
+
+use function mt_rand;
+use function sprintf;
+
+final class Uuid
```
Comment (teghnet) [inline src/Common/Domain/ValueObject/Uuid.php]:
```suggestion
final class Uuid implements \Adshares\Common\Domain\Id
```

**Review by teghnet — CHANGES_REQUESTED** [review]

File: src/Supply/Domain/Model/Banner.php [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    private $id;
+
+    /** @var Campaign */
+    private $campaign;
+
+    /** @var BannerUrl */
+    private $bannerUrl;
+
+    /** @var int */
```
Comment (teghnet) [inline src/Supply/Domain/Model/Banner.php]: int?

File: tests/src/Supply/Application/Service/InventoryImprterTest.php [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+use Adshares\Supply\Application\Service\MarkedCampaignsAsDeleted;
+use Adshares\Supply\Domain\Model\Campaign;
+use Adshares\Supply\Domain\Model\CampaignCollection;
+use Adshares\Supply\Domain\Repository\CampaignRepository;
+use Adshares\Supply\Domain\Repository\Exception\CampaignRepositoryException;
+use Adshares\Supply\Domain\Service\DemandClient;
+use Adshares\Supply\Domain\Service\Exception\EmptyInventoryException;
+use PHPUnit\Framework\TestCase;
+
+final class InventoryImprterTest extends TestCase
```
Comment (PawelPodkalicki) [inline tests/src/Supply/Application/Service/InventoryImprterTest.php]: Typo in class name

**Review by PawelPodkalicki — DISMISSED** [review]

**Review by teghnet — APPROVED** [review]

**Pull request merged**